pkg/lockfile/parse-go-lock.go

```diff
@@ -22,16 +22,25 @@ func ParseGoLock(pathToLockfile string) ([]PackageDetails, error) {
 		return []PackageDetails{}, fmt.Errorf("could not parse %s: %w", pathToLockfile, err)
 	}
 
-	packages := make([]PackageDetails, 0, len(parsedLockfile.Require))
+	packages := map[string]PackageDetails{}
 
 	for _, require := range parsedLockfile.Require {
-		packages = append(packages, PackageDetails{
+		packages[require.Mod.Path+"@"+require.Mod.Version] = PackageDetails{
 			Name:      require.Mod.Path,
 			Version:   strings.TrimPrefix(require.Mod.Version, "v"),
 			Ecosystem: GoEcosystem,
 			CompareAs: GoEcosystem,
-		})
+		}
 	}
 
-	return packages, nil
+	for _, replace := range parsedLockfile.Replace {
+		packages[replace.New.Path+"@"+replace.New.Version] = PackageDetails{
```

Comment:
Should this be referencing `replace.Old` instead in the key? 

The test files only contain `replace`, but if there was a `require` for the old module to be replaced, that'll still  get included right?

Reply:
Correct, per the spec a `require` entry is required:

> Note that a replace directive alone does not add a module to the [module graph](https://go.dev/ref/mod#glos-module-graph). A [require directive](https://go.dev/ref/mod#go-mod-file-require) that refers to a replaced module version is also needed, either in the main module’s go.mod file or a dependency’s go.mod file. A replace directive has no effect if the module version on the left side is not required.